panoramapublic/src/org/labkey/panoramapublic/chromlib/ChromLibStateExporter.java

```diff
@@ -0,0 +1,191 @@
+package org.labkey.panoramapublic.chromlib;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+import org.labkey.api.data.Container;
+import org.labkey.api.security.User;
+import org.labkey.api.targetedms.ITargetedMSRun;
+import org.labkey.api.targetedms.RunRepresentativeDataState;
+import org.labkey.api.targetedms.TargetedMSService;
+import org.labkey.api.writer.PrintWriters;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class ChromLibStateExporter
+{
+    private final Logger _log;
+    private static final String TAB = "\t";
+
+    public ChromLibStateExporter(Logger log)
+    {
+        _log = log;
+    }
+
+    abstract String libTypeString();
+    abstract List<String> headers();
+    abstract void exportLibStateForRun(ITargetedMSRun run, TargetedMSService svc, Logger log, PrintWriter writer, Container container, User user);
+
+    public static void exportLibState(@NotNull File file, @NotNull Container container, User user, Logger log) throws ChromLibStateException
+    {
+        TargetedMSService svc = TargetedMSService.get();
+        TargetedMSService.FolderType folderType = svc.getFolderType(container);
+        if (TargetedMSService.FolderType.LibraryProtein.equals(folderType))
```

Comment:
Optional - use switch statement instead, especially the "enhanced" switch syntax.

https://www.vojtechruzicka.com/java-enhanced-switch/